nodejs/awsx/utils.ts

```diff
@@ -103,3 +105,33 @@ export function hasTrueBooleanMember(obj: any, memberName: string | number | sym
 
     return val === true;
 }
+
+/** @internal */
+export function promiseResult<T>(promise: Promise<T>): T {
+    enum State {
+        running,
+        finishedSuccessfully,
+        finishedWithError,
+    }
+
+    let result: T;
+    let error = undefined;
+    let state = <State>State.running;
```

Comment:
Curious - why is a cast needed?

Reply:
i'm baffled by this as well.  if i remove the cast, TS is now adamant that the local `state` variable has the `.running` value.  it then complains with the `if` check below stating that it's not ever going to be true.  Somehow it doesn't think the assignments in the Promise callbacks would change the flow value of `state`.  I honestly think it's buggy in the compiler/language and want to investigate more.

Reply:
Talked to Daniel!  Unfortunately by design due to https://github.com/microsoft/TypeScript/issues/9998.